M2/Macaulay2/packages/Varieties.m2

```diff
@@ -159,7 +162,8 @@ isWellDefined Variety := X -> (
 	instance(X.cache, CacheTable),
 	"the hash table does not have the expected values")
     -- mathematical checks
-    and assert'(not isProjective X or isHomogeneous R,
+    -- TODO: support non-graded rings (e.g. ZZ, QQ, etc.) and nonstandard gradings
+    and assert'(not isProjective X or isStandardGraded R,
 	"coordinate ring of a projective variety should be homogeneous")
```

Comment:
perhaps the complaint should read "should be homogeneous and generated in degree 1"